server/src/test/scala/org/http4s/server/staticcontent/PathNormalizerSpec.scala

```diff
@@ -1,18 +1,69 @@
 package org.http4s.server.staticcontent
 
-import org.specs2.mutable.Specification
+import java.nio.file.Paths
+import org.scalacheck.Gen
+import org.specs2.ScalaCheck
+import org.specs2.mutable.{Specification, Tables}
 
-class PathNormalizerSpec extends Specification {
+class PathNormalizerSpec extends Specification with Tables with ScalaCheck {
 
   "PathNormalizer.removeDotSegments" should {
-    "remove dot segments correctly" in {
-      val path1 = "/a/b/c/./../../g"
+    // Examples adapted from RFC3986, section 5.4
+    "remove dot segments correctly in RFC examples" >> {
+      "path" | "expected" |>
+        "/a/b/c/./../../g" ! "/a/g" |
+        "mid/content=5/../6" ! "mid/6" |
+        "/a/b/c/./g" ! "/a/b/c/g" |
+        "/a/b/c/g/" ! "/a/b/c/g/" |
+        "/a/b/c/g" ! "/a/b/c/g" |
+        "/a/b/c/../../../g" ! "/g" |
+        "/a/b/c/../../../../g" ! "/g" |
+        "/a/b/c/." ! "/a/b/c/" |
+        "/a/b/c/./" ! "/a/b/c/" |
+        "/a/b/c/.." ! "/a/b/" |
+        "/a/b/c/../" ! "/a/b/" |
+        "/a/b/c/../g" ! "/a/b/g" |
+        "/a/b/c/../.." ! "/a/" |
+        "/a/b/c/../../" ! "/a/" |
+        "/a/b/c/../../g" ! "/a/g" |
+        "/a/b/c/../../../g" ! "/g" |
+        "/a/b/c/../../../../g" ! "/g" |
+        "/a/b/c/g." ! "/a/b/c/g." |
+        "/a/b/c/.g" ! "/a/b/c/.g" |
+        "/a/b/c/g.." ! "/a/b/c/g.." |
+        "/a/b/c/..g" ! "/a/b/c/..g" |
+        "/a/b/c/./../g" ! "/a/b/g" |
+        "/a/b/c/./g/." ! "/a/b/c/g/" |
+        "/a/b/c/g/./h" ! "/a/b/c/g/h" |
+        "/a/b/c/g/../h" ! "/a/b/c/h" |
+        "/a/b/c/g;x=1/./y" ! "/a/b/c/g;x=1/y" |
+        "/a/b/c/g;x=1/../y" ! "/a/b/c/y" | { (path, expected) =>
+        PathNormalizer.removeDotSegments(path) must_=== expected
+      }
+    }
 
-      PathNormalizer.removeDotSegments(path1) must_=== "/a/g"
+    "remove dot segments correctly in other examples" >> prop { input: String =>
```

Comment:
This is rarely going to generate `..` and `.`, or even multiple segments.  I think for this to make sense, we need to have a smarter generator.

And if we do have a property test, ensuring there are no more `./` or `../` would be good assertions.

Reply:
The prop is called with `setGen`, with a custom generator. It's not perfect, but it does create multiple segments and dots.

Reply:
Oh, I missed that the first time. 😬